lib/fs.js

```diff
@@ -1424,6 +1204,8 @@ fs.watch = function(filename, options, listener) {
   if (options.persistent === undefined) options.persistent = true;
   if (options.recursive === undefined) options.recursive = false;
 
+  if (!FSWatcher)
+    ({ FSWatcher } = require('internal/fs/watchers'));
```

Comment:
Not really opposed to this way of using destructuring but it does look a bit odd..if the watchers and streams don't share enough code to be in the same file, maybe just split them into different files, like `internal/fs/event_watcher` for `FSWatcher` and `internal/fs/poll_watcher` for `StatWatcher`? (event/poll is how I differentiate them but using their underscored names is good as well)

Reply:
I'll change the assignments. I'd like to hold off on splitting `StatWatcher` and `FSWatcher` into separate files because I'd like to explore a simplification that could allow them to share code. If it ends up not making sense, I'll split them out in a separate PR.